### PR TITLE
[CI] New CI/CD pipeline and fixes for unit tests

### PR DESCRIPTION
**Pull request by odesenfans**
Added a new pipeline to run unit tests and build the Docker images.
The Docker part is similar to the previous pipeline.

Fixed the unit tests that were easy to fix, and marked the others
as skipped. One important fix is the resolution of a circular
dependency between the chains module and the network module.

Deleted the "demo" pipeline as it served no clear purpose.

### CODE REVIEW

File: .github/workflows/pyaleph-ci.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+jobs:
+  tests:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        # Fetch the whole history for all tags and branches (required for aleph.__version__)
+        fetch-depth: 0
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
```
Comment (hoh) [inline .github/workflows/pyaleph-ci.yml:25]: Is this better than using the distribution's packages via `apt` ?

File: tests/chains/test_register.py [hunk 2]
```diff
@@ -58,7 +56,7 @@ async def test_register_outgoing_worker(monkeypatch):
 
 @pytest.mark.asyncio
 async def test_outgoing():
-    assert len(OUTGOING_WORKERS.keys()) == 3  # 3 verifiers are included by default
+    assert len(OUTGOING_WORKERS) == 2  # 2 verifiers are included by default
```
Comment (hoh) [inline tests/chains/test_register.py:59]: Do you know why this went from 3 to 2 ?

**Review by hoh — COMMENTED** [review]
Looks good. I have 2 minor comments that need to be addressed.

File: .github/workflows/pyaleph-ci.yml (hunk 1 above)
Comment (odesenfans) [reply .github/workflows/pyaleph-ci.yml:25]: Yes, because you can specify the exact version of Python.

File: tests/chains/test_register.py (hunk 2 above)
Comment (odesenfans) [reply tests/chains/test_register.py:59]: The support for BNB was dropped, it was one of the 3.

File: .github/workflows/pyaleph-ci.yml (hunk 1 above)
Comment (hoh) [reply .github/workflows/pyaleph-ci.yml:25]: And in the future test with multiple versions of Python.

**Pull request merged**